pkg/hubble/testutils/payload.go

```diff
@@ -37,6 +36,24 @@ func CreateL3L4Payload(message any, layers ...gopacket.SerializableLayer) ([]byt
 	default:
 		return nil, fmt.Errorf("unsupported message type %T", messageType)
 	}
+
+	// Adjust serialized message length according to event version. This allows us to serialize previous
+	// versions of events in tests, which would be otherwise serialized with the maximum size of the
+	// respective data structure (ex. DropNotifyV1 -> DropNotifyV2 + zero bytes of padding).
+	var offset uint
+	switch messageType := message.(type) {
+	case monitor.DropNotify:
+		offset = messageType.DataOffset()
+	case monitor.TraceNotify:
+		offset = messageType.DataOffset()
+	}
+	if offset != 0 && offset != uint(buf.Len()) {
+		b := buf.Bytes()
+		buf.Reset()
+		buf.Write(b[:offset])
+	}
```

Comment:
Alternatively, we could initially (i.e. around L23) `binary.Encode` into a temporary `[]byte` and then `buf.Write` the correct amount of bytes. No strong feeling either way.

Reply:
Just found the beautiful `buf.Truncate(n)`, I think that's a good compromise 😃